pkg/controller/cronjob/utils.go

```diff
@@ -211,6 +213,12 @@ func getJobFromTemplate2(cj *batchv1.CronJob, scheduledTime time.Time) (*batchv1
 	// We want job names for a given nominal start time to have a deterministic name to avoid the same job being created twice
 	name := getJobName(cj, scheduledTime)
 
+	if utilfeature.DefaultFeatureGate.Enabled(features.CronJobsScheduledAnnotation) {
+
+		// Append job creation timestamp to the cronJob annotations. The time will be in RFC3339 form.
+		annotations[batchv1.CronJobScheduledTimestampAnnotation] = cj.Spec.JobTemplate.CreationTimestamp.Time.Format(time.RFC3339)
```

Comment:
This is still wrong, it should be:
```golang
annotations[batchv1.CronJobScheduledTimestampAnnotation] = scheduledTime.Format(time.RFC3339).In(timezone) 
```
this is because we want to present that annotation in the time that is local to the user, and the information in schedule is transformed to current time zone.

Reply:
It's hard to quickly explain, but when we invoke `cron.ParseStandard` we pass `TZ=<time-zone` which triggers the library to take into account the timezone when calculating the time, but the date returned from invoking `Next` on that schedule will give you time relative to your current time zone. In our case that is the time zone local to the kube-controller-manager. 

It's best to try it locally with something like this:
```golang
sched, _ := cron.ParseStandard("TZ=America/New_York 0 */4 * * *")
next := sched.Next(time.Now())
loc, _ := time.LoadLocation("America/New_York")
fmt.Println("next in TZ", next.In(loc))
fmt.Println("next in local", next)
```
make sure to pick the time zone that is different than your local, the schedule is set to execute every 4h at 0 minutes. 
In my case the above will print (it's currently 21:04pm CEST, which is 15:05 EST):
```
next in TZ 2023-07-06 16:00:00 -0400 EDT
next in local 2023-07-06 22:00:00 +0200 CEST
```
for the annotation we want to present the user with the time printed in TZ.

Reply:
Addressed.

Reply:
> the date returned from invoking Next on that schedule will give you time relative to your current time zone

ah, I see, it uses the location of the time passed in to Next()